client/search/search-input.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useCallback, useImperativeHandle, useRef, useState } from 'react'
+import styled, { css } from 'styled-components'
+import SearchIcon from '../icons/material/baseline-search-24px.svg'
+import { useKeyListener } from '../keyboard/key-listener'
+import { fastOutSlowInShort } from '../material/curves'
+import TextField from '../material/text-field'
+import { useValueAsRef } from '../state-hooks'
+
+const ESCAPE = 'Escape'
+const F = 'KeyF'
+
+const TextFieldContainer = styled(TextField)`
+  ${props =>
```

Comment:
Always prefer using the variable only on the value side of the CSS *unless it cannot be avoided*. So this should be:

```css
width: ${props.isFocused ? 'var(--sb-search-input-focused-width, 250px)' : 'var(--sb-search-input-width, 200px)' };
```